src/protectedApi_v8/user/profile-details.ts

```diff
@@ -419,13 +419,16 @@ profileDeatailsApi.post('/createUser', async (req, res) => {
             if (sbgroup_ === undefined || sbgroup_ === '') {
                 // tslint:disable-next-line: all
                 sbProfileUpdateReq.profileDetails.professionalDetails = _.omit(sbProfileUpdateReq.profileDetails.professionalDetails, 'group')
+                if (sbdesignation_ === undefined || sbdesignation_ === '') {
+                  sbProfileUpdateReq.profileDetails = _.omit(sbProfileUpdateReq.profileDetails, 'professionalDetails')
+                }
```

Comment:
this code is not proper... 
if designation is null -- code removes designation from professional details -- but this may not be possible due to array.

declare profileDetailsObj -- new object -- handle addition of group and designation based on not null... add this new object to profileDetails -- if it has at least one attribute -- otherwise call the update request without professionalDetails in profileDetails..

Reply:
made the changes